department-of-veterans-affairs/caseflow
# Add MDR to CAVC Remand Entry Form

Resolves #15384 / [CASEFLOW-94](https://vajira.max.gov/browse/CASEFLOW-94)

### Description
Fully enables the front end functionality for MDR, locked to the feature flag.

### Acceptance Criteria
- [x] Behind the feature toggle: mdr_cavc_remand
- [x] Accessible to the CAVC Litigation Support Team
- [x] Include screenshot(s) in the Github issue if there are front-end changes
- [x] Update documentation: CAVC Remand
- [x] Verify Memorandum Decision on Remand (MDR) is a radio button on the CAVC entry form
- [x] Verify clicking the MDR radio button shows an infobox under the Court Decision Date picker with the MDR Info copy
- [ ] Verify Confirmation modal is presented on users queue page upon clicking submit on the entry form with the MDR copy

### Testing Plan
  1. Go to (a realistic task tree, dispatched, AMA case)
  2. Hit `+ Add CAVC Remand`
  3. Fill out the form, **skipping Judge selection, judgement date, and mandate date**
  4. Confirm that the infobox appears when MDR is selected
  5. Hit submit - confirm Judgement Date and Mandate Date **do not** display a JS validation error requiring input
  6. Fill out Judge, Judgement Date, and Mandate Date
      - Judgement Date and Mandate date are still required for the _back end_ validation - will be fixed in [CASEFLOW-97](https://vajira.max.gov/browse/CASEFLOW-97)
  7. Hit submit and confirm the success message has the MDR wording, and the CAVC section lists the `Type`: `Memorandum Decision on Remand (MDR)`
      - Again, we are not checking if the MDR flow created the right tasks - that's later in the epic!

Storybook:
  1. Launch storybook and confirm you see the MDR behavior there 
    a. Run `make storybook`
    b. Load AddCavcRemandView under the Queue section
    c. Confirm the MDR radiobox shows, and the infobox appears when its selected

### User Facing Changes

**Info box on MDR selection**
![image](https://user-images.githubusercontent.com/6129479/103581015-2dd05680-4ea9-11eb-8c11-13b9c513737b.png)

**New Success detail**
![image](https://user-images.githubusercontent.com/6129479/103580950-1729ff80-4ea9-11eb-84b7-f73f44c1b0dc.png)


### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Updated existing storybook
